src/common/exif.cc

```diff
@@ -2319,7 +2319,8 @@ int dt_exif_read_blob(uint8_t **buf,
       // GPS data
       dt_remove_exif_geotag(exifData);
       const dt_image_t *cimg = dt_image_cache_get(darktable.image_cache, imgid, 'r');
-      if(!std::isnan(cimg->geoloc.longitude) && !std::isnan(cimg->geoloc.latitude))
+      if(cimg->geoloc.longitude != DT_INVALID_GPS_COORDINATE
```

Comment:
I don't know what the plans are with this and #14143 now that we are getting close to the 4.4 freeze, but even if the actual switch from `NAN` to `-FLT_MAX` isn't activated yet, it would be worth getting all the spread out changes in before the release so that we avoid large changesets _after_ the release that might complicate backporting of bugfixes.

You can still, in `utility.h`,  stick with NAN/isnan() for now and then have a minimal change as the actual switchover. Just have this as a quality-of-codebase PR. For that to work, the few cases, like here and the next few tests, where an actual comparison `!= DT_INVALID_GPS_COORDINATE` is used, rather than `!dt_valid_gps_coordinate()` need to be fixed. I didn't spot any such cases in #14143.

@TurboGit thoughts?

Reply:
@dterrahe : I agree with you that it would be nice to have this in 4.4. So if anything is to be done it should be done soon now.

@ralfbrown : What do you think? Is this plan matching your workload? I certainly don't want to put any more pressure on you :)

I see that you are trying to cope with backward compatibility with DB. I haven't look at the details but I can offer to change value in the DB if needed. But I don't think there is anything to do, as far as I know the longitude/latitude/altitude are either set to valid value or NULL if not defined.

Reply:
> I certainly don't want to put any more pressure on you :)

Absolutely! That's why I suggested maybe leaving the actual value as is for now, if there's not enough time before release, but just changing the way it is tested (in several places) so that it works both ways. And so that the remaining changes are very localized.

Reply:
> I may rework this one to use a one-line test function as in the other PR.

In most places here that work already seems to have been done. There are just the few instances in this file that I pointed out that have been left out. Maybe because this is C++? Would there be a reason why `dt_valid_gps_coordinate` wouldn't be safe here that I'm not aware of? I thought it was just a simple oversight...

Reply:
> My suggestion would be to merge the other PR with flag left at NaN

The other PR is still **Draft**, I let you do a final review and remove the **Draft** flag and I'll merge:
https://github.com/darktable-org/darktable/pull/14143

Reply:
> > I may rework this one to use a one-line test function as in the other PR.
> In most places here that work already seems to have been done.

I'd forgotten that I did that.  And I did indeed simply miss the C++ code due to grepping in *.[ch].

But it turns out that switching the flag value back to NAN introduced a circular inclusion when I needed to include common/math.h for the definition of dt_isnan to make the code safe with -ffinite-math-only.  I unwound that by explicitly including utility.h where it was needed (lots of files) rather than having darktable.h implicitly include it.

@TurboGit Will remove the draft flag after taking another look at the other PR.